helios-parser/src/main/kotlin/helios/parser/AsyncParser.kt

```diff
@@ -160,26 +157,25 @@ class AsyncParser<J>(
 
                         ' ', '\t', '\r' -> offset += 1
                         '[' -> {
-                            if (state == ASYNC_PRESTART) {
-                                offset += 1
-                                state = ASYNC_START
-                            } else if (state == ASYNC_END) {
-                                die(offset, "expected eof")
-                            } else if (state == ASYNC_POSTVAL) {
-                                die(offset, "expected , or ]")
-                            } else {
-                                state = 0
+                            when (state) {
+                                ASYNC_PRESTART -> {
+                                    offset += 1
+                                    state = ASYNC_START
+                                }
+                                ASYNC_END -> die(offset, "expected eof")
+                                ASYNC_POSTVAL -> die(offset, "expected , or ]")
+                                else -> state = 0
                             }
                         }
 
                         ',' -> {
-                            if (state == ASYNC_POSTVAL) {
-                                offset += 1
-                                state = ASYNC_PREVAL
-                            } else if (state == ASYNC_END) {
-                                die(offset, "expected eof")
-                            } else {
-                                die(offset, "expected json value")
+                            when (state) {
```

Comment:
These chnages of `if` for `when` are not valid changes unless we can prove the bytecode emitted is more performant than with `if` statements. This parser is purposedly fast and internally mutable to get the best raw performance out of JSON parsing and speed comes before code style in this case. All other changes look good

Reply:
I'll restore by now and check it after in a separate pr